store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -100,39 +101,66 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
             throw VolumeServiceException.INVALID_REQUEST("Volume Current can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_CURRENT);
         }
 
-        if (enumStoreType.equals(Volume.StoreType.EXTERNAL)) {
-            // validate use in frequent access
-            if (false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() &&
-                true != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess()) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseInFrequentAccess can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS);
-            }
+        if (Volume.StoreType.EXTERNAL.equals(enumStoreType)) {
+            // validate storage type
+            String storageTypeS3 = volInfoRequest.getVolumeExternalInfo().getStorageType();
+            String storageTypeOpenIO = volInfoRequest.getVolumeExternalOpenIOInfo().getStorageType();
 
-            // validate use intelligent tiering
-            if (false != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering() &&
-                true != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering()) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseIntelligentTiering can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_USE_INTELLIGENT_TIERING);
+            if (!storageTypeS3.equalsIgnoreCase(AdminConstants.A_VOLUME_S3) && !storageTypeOpenIO.equalsIgnoreCase(AdminConstants.A_VOLUME_OPEN_IO)) {
+                throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3 or OpenIO", VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
             }
 
-            // validate storage type
-            String storageType = volInfoRequest.getVolumeExternalInfo().getStorageType();
-            if (false == storageType.equals("S3")) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3", VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
+            if(AdminConstants.A_VOLUME_S3.equalsIgnoreCase(storageTypeS3)) {
+                // validate use in frequent access
+                if (false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() &&
+                    true != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess()) {
+                    throw VolumeServiceException.INVALID_REQUEST("Volume UseInFrequentAccess can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS);
+                }
 
-            }
+                // validate use intelligent tiering
+                if (false != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering() &&
```

Comment:
A boolean condition would either true or false, how is it possible it can't be true and false at the same time?

Reply:
again here as well value can be true/false/empty and if anything else should throw exception, so just checking not true Or not false isn't enough add a check for empty as well as below
`if (!StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering()) && false != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering() && true != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering())`

Reply:
Removed the validation, since it is a boolean (not a Boolean), therefore, It can't have any value other than false or true